core/shared/src/test/scala/fs2/ChunkSuite.scala

```diff
@@ -225,13 +228,46 @@ class ChunkSuite extends Fs2Suite {
     Chunk.ArraySlice(Array[Any](0.toByte)).asInstanceOf[Chunk[Byte]].toByteVector
   }
 
-  test("ArraySlice does not copy when chunk is already an ArraySlice instance") {
+  test("toArraySlice does not copy when chunk is already an ArraySlice instance") {
     val chunk: Chunk[Int] = Chunk.ArraySlice(Array(0))
     assert(chunk eq chunk.toArraySlice)
     val chunk2: Chunk[Any] = Chunk.ArraySlice(Array(new Object))
     assert(chunk2 eq chunk2.toArraySlice)
   }
 
+  test("toArraySlice does not copy when chunk is an array-backed bytevector") {
+    val arr = Array[Byte](0)
```

Comment:
The singleton Array may be a degenerate case. Could we use something like `"Hello World".getBytes()`?